python/lsst/meas/algorithms/astrometrySourceSelector.py

```diff
@@ -68,16 +72,24 @@ def selectSources(self, sourceCat, matches=None):
         """
         self._getSchemaKeys(sourceCat.schema)
 
-        result = table.SourceCatalog(sourceCat.table)
-        for source in sourceCat:
-            if self._isGood(source) and not self._isBad(source):
-                result.append(source)
+        if sourceCat.isContiguous():
+            bad = reduce(lambda x, y: np.logical_or(x, sourceCat.get(y)), self.config.badFlags, False)
+            good = self._isGood_vector(sourceCat)
+            result = sourceCat[good & ~bad]
+        else:
+            result = table.SourceCatalog(sourceCat.table)
+            for i, source in enumerate(sourceCat):
+                if self._isGood(source) and not self._isBad(source):
+                    result.append(source)
```

Comment:
I agree, but until we have a generic solution, I'd rather not break non-contiguous functionality.